src/e2e/java/teammates/e2e/pageobjects/InstructorFeedbackSessionsPage.java

```diff
@@ -0,0 +1,552 @@
+    }
+
+    public void verifyResponseRate(FeedbackSessionAttributes session, String expectedResponseRate) {
+        int rowId = getFeedbackSessionRowId(session.getCourseId(), session.getFeedbackSessionName());
+        assertEquals(expectedResponseRate, getResponseRate(rowId));
+    }
+
+    public void addFeedbackSession(FeedbackSessionAttributes newSession, boolean isUsingTemplate) {
+        clickAddSessionButton();
+        waitForElementPresence(By.cssSelector("#instructions iframe"));
+
+        if (isUsingTemplate) {
+            selectDropdownOptionByText(sessionTypeDropdown, "session using template: team peer evaluation");
+        } else {
+            selectDropdownOptionByText(sessionTypeDropdown, "session with my own questions");
+        }
+        selectDropdownOptionByText(courseIdDropdown, newSession.getCourseId());
+        fillTextBox(sessionNameTextBox, newSession.getFeedbackSessionName());
+        setInstructions(newSession.getInstructions());
+        setSessionStartDateTime(newSession.getStartTime(), newSession.getTimeZone());
+        setSessionEndDateTime(newSession.getEndTime(), newSession.getTimeZone());
+        selectGracePeriod(newSession.getGracePeriodMinutes());
+        setVisibilitySettings(newSession);
+        setEmailSettings(newSession);
+
+        clickCreateSessionButton();
+        waitForSessionEditPage();
+    }
+
+    public void addCopyOfSession(FeedbackSessionAttributes sessionToCopy, CourseAttributes copyToCourse,
+                                 String newSessionName) {
+        clickAddSessionButton();
+        click(browser.driver.findElement(By.id("btn-copy-session")));
+
+        selectCourseToCopy(copyToCourse.getId());
+        fillTextBox(browser.driver.findElement(By.id("copy-session-name")), newSessionName);
+        selectSessionToCopy(sessionToCopy.getCourseId(), sessionToCopy.getFeedbackSessionName());
+
+        clickConfirmCopySessionButton();
+        waitForSessionEditPage();
+    }
+
+    public void copySession(FeedbackSessionAttributes sessionToCopy, CourseAttributes copyToCourse,
+                            String newSessionName) {
+        String copyFromCourse = sessionToCopy.getCourseId();
+        String sessionName = sessionToCopy.getFeedbackSessionName();
+        WebElement copyFsModal = clickCopyButtonInTable(copyFromCourse, sessionName);
+
+        fillTextBox(copyFsModal.findElement(By.id("copy-session-name")), newSessionName);
+        selectCourseToCopyToInModal(copyFsModal, copyToCourse.getId());
+
+        click(browser.driver.findElement(By.id("btn-confirm-copy-course")));
+    }
+
+    public void moveToRecycleBin(FeedbackSessionAttributes sessionToDelete) {
+        int rowId = getFeedbackSessionRowId(sessionToDelete.getCourseId(), sessionToDelete.getFeedbackSessionName());
+        clickAndConfirm(browser.driver.findElement(By.id("btn-soft-delete-" + rowId)));
+        ThreadHelper.waitFor(1000);
```

Comment:
What are all these waits for?

Reply:
It is actually for the animation time when the tables are created. I have changed to using the `waitUntilAnimationFinish` method that someone else wrote.